File: app/src/main/java/com/zurche/topreddit/home/service/RedditService.java
package com.zurche.topreddit.home.service;

import com.zurche.topreddit.home.service.model.TopStoriesResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @author alejandro.zurcher
 * @since {insert app current main version here}
 */

public interface RedditService {
    @GET("top.json")
    Call<TopStoriesResponse> listTopStories(@Query("limit") String limit);
}

File: app/src/main/java/com/zurche/topreddit/home/data/remote/service/RedditService.kt
package com.zurche.topreddit.home.data.remote.service

import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author alejandro.zurcher
 * @since {insert app current main version here}
 */
interface RedditService {

    @GET("top.json")
    suspend fun listTopStoriesAfter(@Query("limit") pageSize: Int,
                                    @Query("after") currentPage: String?)
            : Response<TopStoriesResponse>
}